.github/workflows/ci.yml

```diff
@@ -3,7 +3,8 @@ name: CI
 on:
   push:
     branches:
-      - '*'
+      - 'main'
+      - '*.*'
```

Comment:
This should only catch branches that have "versions" aka 0.1 or 1.0 etc.
If you plan to maintain a release branch.